src/plugins/announce.lua

```diff
@@ -0,0 +1,53 @@
+-- URL announcements - follows http->https redirects but not www
+
+local https = require 'ssl.https'
+local ltn12 = require 'ltn12'
+
+local plugin = {}
+
+plugin.help = 'Usage: announce <url>'
+
+plugin.main = function(args)
+    local _, _, url = args.message:find('(.+)')
+
+    if args.modules.config.debug then
+        print(url)
+    end
+
+    if not url then
+        args.modules.irc.privmsg(args.target, ('%s: Please give me a url.'):format(args.sender))
+        return
+    end
+
+    if url:find('[ <]?https?://[^> ]+.*') then
+        url = url:match('[ <]?https?://[^> ]+.*')
+    elseif url:find(' www%..*') then
```

Comment:
Given the above change, I think this `elseif` block is unnecessary. This makes it so that people need to either specify the URL properly somewhere in their message, or have the whole message just be the URL (which will be the `else` block).

Reply:
should be done :dancer: